GameWorld/View3D/Services/SceneSaving/Material/Strategies/WsModelGeneratorService.cs

```diff
@@ -77,19 +78,39 @@ string CreateWsModel(MainEditableNode mainNode, GameTypeEnum game, string modelF
             sb.Append("\t\t<materials>\n");
 
             var lodNodes = mainNode.GetLodNodes();
-            for (var lodIndex = 0; lodIndex < lodNodes.Count; lodIndex++)
+            if (game != GameTypeEnum.Pharaoh)
             {
-                var meshes = mainNode.GetMeshesInLod(lodIndex, false);
-                var uniqueNames = GenerateUniqueNames(meshes);
-                for (var meshIndex = 0; meshIndex < meshes.Count; meshIndex++)
+                for (var lodIndex = 0; lodIndex < lodNodes.Count; lodIndex++)
                 {
-                    var materialFile = GetOrCreateMaterial(modelFilePath, game, meshes[meshIndex], uniqueNames[meshIndex], materialTemplate);
-                    sb.Append($"\t\t\t<material lod_index=\"{lodIndex}\" part_index=\"{meshIndex}\">");
-                    sb.Append(materialFile);
-                    sb.Append("</material>\n");
+                    var meshes = mainNode.GetMeshesInLod(lodIndex, false);
```

Comment:
The two different paths here are almost identical. The if should change as little as possible. 
A suggestion would be to add a "MaterialIndexName" to the wsmodel strategy interface. So the strategy can return the value, then there is no need for a if at all